### PR TITLE
Odo v3: Move odo registry to odo preference registry, remove unused pref. settings

### PR DESCRIPTION
**Pull request by cdrage**
**What type of PR is this:**

/kind feature

**What does this PR do / why we need it:**

This PR does the following:
- Moves `odo registry` to `odo preference registry`
- Gets rid of unused preference configuration options BuildTimeout and NamePrefix
- Reorders the parameters for preference for the usage in --help
- Adds RegistryCacheTime as a parameter that you can set (it was previously hidden before)
- Adds more description with regarding to parameters

**Which issue(s) this PR fixes:**

https://github.com/redhat-developer/odo/issues/5402

**PR acceptance criteria:**

- [X] Unit test

- [X] Integration test

- [X] Documentation

**How to test changes / Special notes to the reviewer:**

### CODE REVIEW

**Comment by cdrage** [comment]
Ready for reviews! No longer WIP and I've updated the description / added more tests.

@feloy @dharmit @kadel @anandrkskd

File: pkg/component/component.go [hunk 1]
```diff
@@ -70,17 +70,18 @@ func GetDefaultComponentName(cfg preference.Client, componentPath string, compon
...
-	if cfg.NamePrefix() == nil || *cfg.NamePrefix() == "" {
-		prefix, err = GetComponentDir(componentPath)
-		if err != nil {
-			return "", errors.Wrap(err, "unable to generate random component name")
-		}
-		prefix = util.TruncateString(prefix, componentRandomNamePartsMaxLen)
-	} else {
-		// Set the required prefix into componentName
-		prefix = *cfg.NamePrefix()
+	// V2 Change -> V3
```
Comment (feloy) [inline pkg/component/component.go:73]: Could you please remove this commented code?

File: pkg/odo/cli/preference/unset.go [hunk 2]
```diff
@@ -92,9 +92,12 @@ func NewCmdUnset(name, fullName string) *cobra.Command {
 		Short: "Unset a value in odo preference file",
 		Long:  fmt.Sprintf(unsetLongDesc, preference.FormatSupportedParameters()),
 		Example: func(exampleString, fullName string) string {
-			for _, property := range preference.GetSupportedParameters() {
-				exampleString += fmt.Sprintf("\n  %s %s", fullName, property)
-			}
+			// For now, the --help page is too large listing all unset values too.
+			// Just show one example of how to unset a value.
+			//for _, property := range preference.GetSupportedParameters() {
```
Comment (feloy) [inline pkg/odo/cli/preference/unset.go:97]: I'm not sure it is necessary to keep old code here. You could remove the code, and explain why we only display for 1st  parameter.

**Review by feloy — COMMENTED** [review]
Somme remarks about comments, otherwise lgtm

**Comment by cdrage** [comment]
Thanks for the comments @feloy I've updated the PR as well as made changes based on your feedback.

**Comment by feloy** [comment]
/test v4.9-integration-e2e

Infra errors

**Comment by feloy** [comment]
/test v4.9-integration-e2e

```
        cmd [ps -ef] failed with error command terminated with exit code 1
```

**Comment by feloy** [comment]
/test v4.9-integration-e2e

```
[Fail] odo devfile supported tests odo debug support for devfile components [It] Verify output debug information for java-springboot works 
```

**Pull request merged**